packages/lit-dev-content/site/blog/2022-02-07-eleventy.md

```diff
@@ -0,0 +1,181 @@
+Eleventy Node process, and evaluating the element's template using a special
+version of Lit's `render` function. This special `render` function behaves much
+like the `render` function that Lit uses in the browser, except that it emits
+stream of strings, instead of writing to the DOM.
+
+For more information about `@lit-labs/ssr`, checkout its [GitHub
+repo](https://github.com/lit/lit/tree/main/packages/labs/ssr). We'll be
+expanding the documentation soon to help developers integrate Lit pre-rendering
+into their tools and frameworks too.
+
+### Declarative Shadow DOM
+
+When a component is statically pre-rendered, its shadow DOM and styles are
+represented using the `<template shadowroot>` HTML element. When the browser's
+HTML parser encounters one of these elements, it applies the template's content
+as the shadow root of its parent element. This way, all of the DOM and style
+encapsulation guarantees of web components are preserved, all using declarative
+HTML.
+
+As of February 2022, Chrome and Edge have native support for Declarative Shadow
+DOM, but Firefox and Safari have not yet implemented it. A very small polyfill
+is available for browsers that don't yet have support. Integrating the polyfill
+into your site is
+[documented](https://github.com/lit/lit/tree/main/packages/labs/eleventy-plugin-lit#ponyfill)
+in the `@lit-labs/eleventy-plugin-lit` README.
+
+For more information about Declarative Shadow DOM, check out *[Declarative
+Shadow DOM (web.dev)](https://web.dev/declarative-shadow-dom/)*.
+
+### Re-hydration
+
+While static pre-rendering lets your components initial paint happen as quickly
+as possible, the components won't yet be interactive, because their JavaScript
+implementations haven't loaded.
+
+*Re-hydration* is the process where a statically pre-rendered component is
+upgraded to its JavaScript implementation, becoming responsive and interactive.
+
+Lit comes with automatic support for re-hydration via the special
+`lit/experimental-hydrate-support.js` module. As long as this module has been
+loaded, Lit components will detect when they have been pre-rendered, and adopt
+the shadow root that has already been attached to them.
+
+Re-hydration with Lit is very efficient, because Lit knows that it doesn't need
+to re-render the existing shadow root when it upgrades. If any data has changed
+between pre-rendering and upgrade, then only the parts of the shadow root that
+changed will be updated.
+
+## What's next?
+
+Check out the
+[roadmap](https://github.com/lit/lit/tree/main/packages/labs/eleventy-plugin-lit#roadmap)
+for `@lit-labs/eleventy-plugin-lit` for some of the main features and fixes you
+can expect to land over the coming months.
+
+If you try out `@lit-labs/eleventy-plugin-lit`, let us know if you have any
+suggestions or find any problems by starting a
+[discussion](https://github.com/lit/lit/discussions) or filing an
+[issue](https://github.com/lit/lit/issues).
```

Comment:
Doesn't need to be addressed, but we could also add some query params here so that we can quickly identify where issues have come from. I.e.

`https://github.com/lit/lit/issues/new?title=[11ty-ssr]:%20Describe%20your%20issue` Will pre-fill a title with [11ty-ssr].

Reply:
That's a great thought but I feel like linking to `/new` bypasses the template selection and skips the user potentially searching for an existing issue.

Reply:
Yeah, I wanted to encourage searching for existing issues.